bolero/environment/optimum_trajectory.py

```diff
@@ -294,5 +294,116 @@ def plot(self, ax):
         if self.obstacles is not None:
             from matplotlib.patches import Circle
             for obstacle in self.obstacles:
-                ax.add_patch(Circle(obstacle, self.obstacle_dist, ec="none",
+                ax.add_patch(Circle(np.asarray(obstacle).copy(), self.obstacle_dist, ec="none",
                                     color="r"))
+
+
+class OptimumTrajectoryCurbingObstacles(OptimumTrajectory):
+    """Optimize a trajectory according to some criteria. Passing by obstacles slows down or stops the movement
```

Comment:
The first sentence should describe the whole class briefly.

Reply:
You mean it should only be one sentence in the first line? Here's an adapted version:
**"Optimize a trajectory according to some criteria while movements are slowed down or stopped in the vicinity of obstacles"**

Reply:
> You mean it should only be one sentence in the first line?

Yes, that is what I meant. That suggestion is too long though. :)

What about "Search trajectories with several criteria and curbing near obstacles.", followed by a more detailed description in the next paragraph?

Reply:
Good. Reflecting our decision on the [non-point nature](https://github.com/rock-learning/bolero/pull/69#discussion_r219422851) of obstacles, I propose
"Search trajectories with several criteria and curbing *when passing through* obstacles."

Reply:
the pep8 79 char limit made me rethink: 
"Search trajectories with several criteria and curbing within obstacles."